spec/javascripts/collectionBinder.spec.js

```diff
@@ -0,0 +1,311 @@
+describe("CollectionBinder", function () {
+    describe("Without an elManagerFactory specified", function () {
+        it("should throw exception", function () {
+            expect(function () { new Backbone.CollectionBinder(); })
+                .toThrow();
+        });
+    });
+
+    describe("bind", function () {
+        describe("Without the necessary options", function () {
+            beforeEach(function () {
+                var elManagerFactory = new Backbone.CollectionBinder.ElManagerFactory("<span></span>");
+                this.target = new Backbone.CollectionBinder(elManagerFactory);
+            });
+
+            it("should throw exception if collection not provided", function () {
+                expect(function () { this.target.bind(); })
+                    .toThrow();
+            });
+
+            it("should throw if parent element not provided", function () {
+                var collection = new Backbone.Collection();
+
+                expect(function () { this.target.bind(collection) })
+                    .toThrow();
+            });
+        });
+
+        describe("Using collection events", function () {
```

Comment:
I would remove this section. It doesn't test any behavior expected by a user.

Instead we should test that CollectionBinder adds elements on collection.add, remove on collection.remove, do complex actions on collection.set/reset.